### ISSUE TITLE
Creating Physical Component 

### ISSUE THREAD

**Issue by Straniero95** [DSD-DBS/py-capellambse#484]
Hello! I was trying to create a PhysicalComponent by using capellambse. 

Here's the code I'm using: 
```

        import capellambse
        from pathlib import Path
        
        def create_physical_component(model_path):
            # Upload model
            model = capellambse.model.MelodyModel(model_path)
        
            # Access to the Physical Architecture
            physical_architecture = model.pa
        
            # Find the parent component for the new one
            parent_component = physical_architecture.all_components.by_name("Physical System")
            if not parent_component:
                raise ValueError("Parent component 'Physical System' not foundd.")
        
            # Create a new PhysicalComponent as owned_component of the parent component
            new_physical_component = parent_component.owned_components.create(
                name="New component 1",
                description="This component has be created with Python."
            )
        
            new_physical_component.nature   = 'NODE'
            new_physical_component.kind     = 'UNSET'
        
            # Apply PVMT domains and groups
            for domain in model.pvmt.domains:
                print(f"Processing domain: {domain.name}")
                for group in domain.groups:
                    if group.applies_to(new_physical_component):
                        group.apply(new_physical_component)
                        print(f"Applied group '{group.name}' from domain '{domain.name}'.")
        
            # Assign the property
            for groupdef in new_physical_component.pvmt.groupdefs:
                for prop_val in groupdef.property_values:
                    if prop_val.name == 'Property':
                        literals = prop_val.value.pvmt.owner.parent.literals
                        target_literal = next((lit for lit in literals if lit.name == '2'), None)
                        if target_literal:
                            prop_val.value = target_literal
                            print(f"Set 'Property' to '{target_literal.name}' in domain '{groupdef.fullname}'.")
        
            # Save the model
            model.save()
        
            print(f"Physical Component '{new_physical_component.name}' created successfully!")
        
        path = str(Path(__file__).resolve())
        # .aird file path
        capella_path = "\\path\\"
        
        model_name = "file.aird"
        model_path = path + capella_path + model_name
        create_physical_component(model_path)_
```

As I open the Capella model, the component is created, but It's not related to the parent_component and as soon as I try to see the Properties I get the following error: 
"org.polarsys.capella.core.data.capellacore.EnumerationPropertyType.getOwnedLiterals()" because the return value of "org.polarsys.capella.core.data.capellacore.EnumerationPropertyValue.getType()" is null

I've been trying searching for a solution, but I found nothing. Can you help me?

**Comment by Wuestengecko** [comment]
Looks like there's an issue with how `PropertyValueGroup.apply()` deals with Enumerations. I'll try to figure this out and come up with a fix.

**Comment by Wuestengecko** [comment]
Could you please re-test with the changes from #487? You can install it with:

```
pip install git+https://github.com/DSD-DBS/py-capellambse@pvmt-enum-fix
```

(Use `pip install -U capellambse` to return to the latest release.)

**Comment by Straniero95** [comment]
Thanks! I managed to solve the problem.

However, may I ask why the component is not linked to its parent immediately after creation? When I enter the Capella diagram, the component appears, but the link to the parent is missing and cannot be established manually. Interestingly, the link works only when I copy and paste the component.

From what I’ve found, the issue seems to be related to the Part, which is not automatically created when using the owned_components.create function.

**Comment by Wuestengecko** [comment]
> From what I’ve found, the issue seems to be related to the Part, which is not automatically created when using the owned_components.create function.

Yes, that's exactly right. And curiously, I also found that we don't even properly expose Parts yet, so I went ahead and added that.

With the just released capellambse v0.6.11, that is now possible, although still has to be done explicitly. You can do so like this:

```python
    ...
    # Create a new PhysicalComponent as owned_component of the parent component
    new_physical_component = parent_component.owned_components.create(
        name="New component 1",
        description="This component has be created with Python."
    )

    new_physical_component.nature   = 'NODE'
    new_physical_component.kind     = 'UNSET'

    # Create a Part for the new PhysicalComponent
    parent_component.owned_features.create(
        "Part", name=new_physical_component.name, type=new_physical_component
    )

    # Apply PVMT domains and groups
    ...
```

**Comment by tkSDISW** [comment]
> > From what I’ve found, the issue seems to be related to the Part, which is not automatically created when using the owned_components.create function.
> 
> Yes, that's exactly right. And curiously, I also found that we don't even properly expose Parts yet, so I went ahead and added that.
> 
> With the just released capellambse v0.6.11, that is now possible, although still has to be done explicitly. You can do so like this:
> 
>     ...

I tried this with Physical and Logical components and had no luck. Having the support to work with YAML would really be helpful. 

`
--------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[2], line 223
    220 row = excel_content.get_first_row()
    221 current_level = 0
--> 223 Create_Component_Level(current_level,parent,excel_content)
    225 #model.save()

Cell In[2], line 188, in Create_Component_Level(current_level, parent, excel_content)
    185 print(parent)
    186 new_lc = parent.owned_components.create(name = excel_content.get_current_row()['Title'] )
--> 188 parent.owned_features.create("Part", name= new_lc.name, type=new_lc)          
    189 print("created :", excel_content.get_current_row()["Title"])
    190 excel_content.get_next_row() 

File /usr/local/lib/python3.11/dist-packages/capellambse/model/_obj.py:1315, in ElementListCouplingMixin.create(self, typehint, **kw)
   1313 acc = type(self)._accessor
   1314 assert isinstance(acc, _descriptors.WritableAccessor)
-> 1315 newobj = acc.create(self, typehint, **kw)
   1316 try:
   1317     acc.insert(self, len(self), newobj)

File /usr/local/lib/python3.11/dist-packages/capellambse/model/_descriptors.py:1864, in Containment.create(self, elmlist, typehint, **kw)
   1857 def create(
   1858     self,
   1859     elmlist: _obj.ElementListCouplingMixin,
   (...)
   1862     **kw: t.Any,
   1863 ) -> _obj.ModelElement:
-> 1864     return self._create(elmlist._parent, self.role_tag, typehint, **kw)

File /usr/local/lib/python3.11/dist-packages/capellambse/model/_descriptors.py:331, in WritableAccessor._create(self, parent, xmltag, typehint, **kw)
    322 def _create(
    323     self,
    324     parent: _obj.ModelObject,
   (...)
    328     **kw: t.Any,
    329 ) -> T_co:
    330     if typehint:
--> 331         elmclass, kw["xtype"] = self._match_xtype(typehint)
    332     else:
    333         elmclass, kw["xtype"] = self._guess_xtype()

File /usr/local/lib/python3.11/dist-packages/capellambse/model/_descriptors.py:1931, in Containment._match_xtype(self, hint)
   1929 cls = next((i for i in self.classes if i.__name__ == hint), None)
   1930 if cls is None:
-> 1931     raise ValueError(f"Invalid class for {self._qualname}: {hint}")
   1932 return cls, _xtype.build_xtype(cls)

ValueError: Invalid class for Component.owned_features: Part
`
